spring-integration-file/src/main/java/org/springframework/integration/file/remote/AbstractRemoteFileStreamingMessageSource.java

```diff
@@ -74,6 +79,12 @@ protected AbstractRemoteFileStreamingMessageSource(RemoteFileTemplate<F> templat
 			Comparator<AbstractFileInfo<F>> comparator) {
 		this.remoteFileTemplate = template;
 		this.comparator = comparator;
+		if (JacksonJsonUtils.isJackson2Present()) {
+			this.objectMapper = JsonObjectMapperProvider.newInstanceBuilder(false).build();
```

Comment:
So, if there is no `jackson`, but is `boon` we fail with the:

> Neither jackson-databind.jar, nor boon.jar 

Because of:
```
else if (boonPresent) {
	return new BoonJsonObjectMapperBuilder();
}
else {
	throw new IllegalStateException("Neither jackson-databind.jar, nor boon.jar is present in the classpath.");
}
```

Why an approach from the `FileSplitter` doesn't work here?
```
JsonObjectMapperProvider.jsonAvailable() ? JsonObjectMapperProvider.newInstance() : null;
```
In other words fallback to `boon`  if present and no `jackson`.